hw1/server/src/authenticateUser.ts

```diff
@@ -0,0 +1,35 @@
+import { users } from "./assets/users";
+import { User } from "./types/UserType";
+
+class AuthenticateUser {
+
+    private username: string;
+    private password: string;
+
+    constructor(username: string, password: string) {
+
+        this.username = username;
+        this.password = password;
+
+    }
+
+   public auth(): Promise<User> {
+
+        return new Promise((res, req) => {
```

Comment:
Promise cb take (resolve, reject) params . (res and req) - shorts from request / response  this is not for promiser. we usually use (req,res) params in function that handle request